ipld/kamt/src/iter.rs

```diff
@@ -0,0 +1,156 @@
+// Copyright 2021-2023 Protocol Labs
+// SPDX-License-Identifier: Apache-2.0, MIT
+use std::borrow::Borrow;
+use std::iter::FusedIterator;
+
+use fvm_ipld_blockstore::Blockstore;
+use fvm_ipld_encoding::de::DeserializeOwned;
+use fvm_ipld_encoding::CborStore;
+
+use crate::hash_bits::HashBits;
+use crate::node::{match_extension, ExtensionMatch, Node};
+use crate::pointer::Pointer;
+use crate::{AsHashedKey, Config, Error, KeyValuePair};
+#[doc(hidden)]
+pub struct IterImpl<'a, BS, V, K, H, const N: usize = 32> {
+    store: &'a BS,
+    stack: Vec<std::slice::Iter<'a, Pointer<K, V, H, N>>>,
+    current: std::slice::Iter<'a, KeyValuePair<K, V>>,
+}
+
+pub type Iter<'a, BS, V, K, H, const N: usize = 32> = IterImpl<'a, BS, V, K, H, N>;
```

Comment:
We don't need a separate `Iter` and `IterImpl`. We only have that for the `Hamt` to handle the multiple versions.